### PR TITLE
Add VoltageRegulation extension

### PR DESCRIPTION
**Pull request by colinepiloquet**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

No

**What kind of change does this PR introduce?**

Feature

**What is the new behavior (if this is a feature change)?**
New extension: VoltageRegulation

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [x] No

### CODE REVIEW

File: iidm/iidm-serde/src/main/java/com/powsybl/iidm/serde/extensions/VoltageRegulationSerDe.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+import java.util.Objects;
+
+/**
+ * @author Coline Piloquet {@literal <coline.piloquet@rte-france.fr>}
+ */
+@AutoService(ExtensionSerDe.class)
+public class VoltageRegulationSerDe<T extends Injection<T> & ReactiveLimitsHolder> extends AbstractVersionableNetworkExtensionSerDe<T, VoltageRegulation<T>> {
+
+    public VoltageRegulationSerDe() {
+        super("voltageRegulation", VoltageRegulation.class, "vr",
```
Comment (colinepiloquet) [inline iidm/iidm-serde/src/main/java/com/powsybl/iidm/serde/extensions/VoltageRegulationSerDe.java:36]: I don't know if we want it just for IIDM 1.12/ if we can add it for the other versions. Also do we want in core the second version of the extension for IIDM 1.0?

Comment (annetill) [reply iidm/iidm-serde/src/main/java/com/powsybl/iidm/serde/extensions/VoltageRegulationSerDe.java:36]: @geofjamg help :-)

File: iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/extensions/VoltageRegulationImpl.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        }
+    }
+
+    @Override
+    public Terminal getRegulatingTerminal() {
+        return regulatingTerminal;
+    }
+
+    @Override
+    public void setRegulatingTerminal(Terminal regulatingTerminal) {
```
Comment (flo-dup) [inline iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/extensions/VoltageRegulationImpl.java:56]: Just thinking at the behaviour of the SlackTerminal extension... It's strange to support multivariant and have a setter here but not supporting multivariant for that parameter.

**Review by flo-dup — APPROVED** [review]

File: iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/extensions/VoltageRegulationImpl.java (hunk 2 above)
Comment (annetill) [reply iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/extensions/VoltageRegulationImpl.java:56]: The regulating terminal is not variant dependent...

**Pull request merged**